src/compiler/plugin.ts

```diff
@@ -36,24 +36,48 @@ export class Version extends pb_1.Message {
     set major(value: number) {
         pb_1.Message.setField(this, 1, value);
     }
+    clear_major() {
+        pb_1.Message.setField(this, 1, undefined);
+    }
+    has_major() {
+        return pb_1.Message.getField(this, 1) != null;
+    }
     get minor() {
         return pb_1.Message.getFieldWithDefault(this, 2, 0) as number;
     }
     set minor(value: number) {
         pb_1.Message.setField(this, 2, value);
     }
+    clear_minor() {
+        pb_1.Message.setField(this, 2, undefined);
+    }
+    has_minor() {
+        return pb_1.Message.getField(this, 2) != null;
+    }
     get patch() {
         return pb_1.Message.getFieldWithDefault(this, 3, 0) as number;
     }
     set patch(value: number) {
         pb_1.Message.setField(this, 3, value);
     }
+    clear_patch() {
```

Comment:
I wasn't aware that we needed this. How's this any different than setting `patch = undefined`?

Reply:
This is an internal project file, and the whole project seems to be compiling from typescript to javascript without `strictNullChecks` option of `tsc` enabled.  So I guess the `clear_patch()` method is really unnecessary here.

On the other hand, for users of protoc-gen-ts who have the `strictNullChecks` option enabled in their projects the existence of the `clear_property()` methods makes sense. With `strictNullChecks` enabled the `tsc` compiler will not allow the assignment of undefined, like `patch = undefined` in this example, simply because the setter accepts a value of some specific type, other than `undefined` (a `number` in this example.)